File: finstack-py/tests/golden/runners/pricing_equity_index_future.py
"""Domain runner for `equity.equity_index_future` golden fixtures."""

from __future__ import annotations

from tests.golden.schema import GoldenFixture

from .pricing_common import run_pricing_fixture


def run(fixture: GoldenFixture) -> dict[str, float]:
    """Run one equity index future fixture through the Python bindings."""
    return run_pricing_fixture(fixture)

File: finstack/valuations/tests/golden/runners/pricing_equity_index_future.rs
//! Domain runner for `equity.equity_index_future` golden fixtures.

use crate::golden::runner::DomainRunner;
use crate::golden::schema::GoldenFixture;
use std::collections::BTreeMap;

/// Equity index future pricing golden runner.
pub struct EquityIndexFutureRunner;

impl DomainRunner for EquityIndexFutureRunner {
    fn run(&self, fixture: &GoldenFixture) -> Result<BTreeMap<String, f64>, String> {
        crate::golden::runners::pricing_common::run_pricing_fixture(fixture)
    }
}
